browser/ui/webui/chat_ui/chat_ui.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/chat_ui/chat_ui.h"
+
+#include <utility>
+
+#include "brave/browser/ui/webui/brave_webui_source.h"
+#include "brave/browser/ui/webui/chat_ui/chat_ui_page_handler.h"
+#include "brave/components/chat_ui/common/chat_ui_constants.h"
+#include "brave/components/chat_ui/common/pref_names.h"
+#include "brave/components/chat_ui/resources/page/grit/chat_ui_generated_map.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/components/l10n/common/localization_util.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/webui/webui_util.h"
+#include "components/grit/brave_components_resources.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "content/public/common/url_constants.h"
+#include "services/network/public/mojom/content_security_policy.mojom.h"
+
+ChatUI::ChatUI(content::WebUI* web_ui)
+    : ui::UntrustedWebUIController(web_ui),
+      profile_(Profile::FromWebUI(web_ui)) {
+  DCHECK(profile_);
+  browser_ = chrome::FindLastActiveWithProfile(profile_);
+
+  // Create a URLDataSource and add resources.
+  content::WebUIDataSource* untrusted_source =
+      content::WebUIDataSource::CreateAndAdd(
+          web_ui->GetWebContents()->GetBrowserContext(), kChatUIURL);
+
+  webui::SetupWebUIDataSource(
+      untrusted_source, base::make_span(kChatUiGenerated, kChatUiGeneratedSize),
+      IDR_CHAT_UI_HTML);
+
+  for (const auto& str : chat_ui::kLocalizedStrings) {
+    std::u16string l10n_str =
+        brave_l10n::GetLocalizedResourceUTF16String(str.id);
+    untrusted_source->AddString(str.name, l10n_str);
+  }
+
+  untrusted_source->AddBoolean(
+      "hasSeenAgreement",
+      profile_->GetOriginalProfile()->GetPrefs()->GetBoolean(
+          chat_ui::prefs::kBraveChatHasSeenDisclaimer));
+
+  untrusted_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ScriptSrc,
+      "script-src 'self' chrome-untrusted://resources;");
+  untrusted_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::StyleSrc,
+      "style-src 'self' 'unsafe-inline' chrome-untrusted://resources;");
```

Comment:
does this really need unsafe-inline styles? can styles be moved to separate files?

Reply:
f I remove unsafe-inline here, I get the following errors in the console. It looks like this function comes from Webpack's style loader plugin, which we use for .s/css extensions in our JS. Some of these functions add styles to the DOM directly.

@petemill Any tips on how to solve this?

<img width="628" alt="image" src="https://user-images.githubusercontent.com/8665427/228407484-18d9d3e3-d224-4d42-80e5-7ff534838125.png">

Reply:
that sounds ok